lib/rubocop/cop/style/yoda_condition.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for Yoda conditions where the two parts of an expression
+      # are reversed from the typical order in a conditional statement.
+      #
+      # @example
+      #
+      #   # bad
+      #   99 == foo
+      #   "bar" == foo
+      #
+      # @example
+      #
+      #   # good
+      #   foo == 99
+      #   foo == "bar"
+      class YodaCondition < Cop
+        MSG = 'Reverse the order in conditional statement: `%s` == `%s`.'.freeze
+
+        def on_send(node)
+          return unless comparison_operator?(node)
+
+          register_offense(node) if yoda_condition?(node)
```

Comment:
Code looks really good now!

I would probably reorganize the conditions a bit. Currently, the `#yoda_condition?` method doesn't actually check if something is a Yoda condition. I would wrap the check for comparison operator and whitelist types in that method and then do this in the callback:

```ruby
return unless yoda_condition?(node)
```